### PR TITLE
Fix search

### PR DESCRIPTION
**Pull request by emumoo**
@aw3as, @keenadams search is better now

ended up adjusting professor page as well
and added some back links to course show page

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -244,6 +244,9 @@ select {
   margin-bottom: 0px;
   font-weight: 300;
 }
+.title-link:hover {
```
Comment (aw3as) [inline app/assets/stylesheets/application.scss]: what's this referring to?

File: app/assets/javascripts/scheduler.js [hunk 2]
```diff
@@ -232,8 +229,9 @@ $(document).ready(function() {
 			$('.ui-autocomplete').css('width', ($('#class-search').parent().width()));
 		},
 		select: function(event, ui) {
-			courseSearch(ui.item.value);
-		}
+			courseSearch(ui.item.label);
+		},
+		minLength: 2
```
Comment (aw3as) [inline app/assets/javascripts/scheduler.js]: move this field to the top - put function handlers on bottom

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -271,3 +274,9 @@ select {
   }
 }
 
+.ui-autocomplete {
```
Comment (aw3as) [inline app/assets/stylesheets/application.scss]: glad we found the styling for this

File: app/assets/stylesheets/professor.scss [hunk 4]
```diff
@@ -1,3 +1,45 @@
 @import "variables";
```
Comment (aw3as) [inline app/assets/stylesheets/professor.scss]: do we even use this page??

File: app/controllers/scheduler_controller.rb [hunk 5]
```diff
@@ -35,12 +35,22 @@ def course
   end
 
   def search
-    courses = Course.where('title LIKE ?', "%#{params[:term]}%")
-    courses += Section.where('topic LIKE ?', "%#{params[:term]}%").map(&:course).uniq
+    @query = params[:query].strip.split(' ')
+    courses = []
+    if @query.length != 1 and @query[1].to_i.to_s == @query[1]
```
Comment (aw3as) [inline app/controllers/scheduler_controller.rb]: use regex here

File: app/controllers/scheduler_controller.rb [hunk 6]
```diff
@@ -35,12 +35,22 @@ def course
...
+      courses += Course.includes(:subdepartment).current.where('subdepartments.mnemonic LIKE ? AND course_number LIKE ?', "%#{@query[0]}%", "%#{@query[1]}%").references(:subdepartment)
+    end
+    
+    if @query[0].to_i.to_s == @query[0]
```
Comment (aw3as) [inline app/controllers/scheduler_controller.rb]: regex pl0x

File: app/models/course.rb [hunk 7]
```diff
@@ -23,12 +23,21 @@ class Course < ActiveRecord::Base
...
+    Course.where(:last_taught_semester_id => semester_id)
+  end
+
   def professors_list
     self.professors.uniq{ |p| p.id }.sort_by{|p| p.last_name}
   end
 
   def mnemonic_number
-    @mnemonic_number ||= "#{subdepartment.mnemonic} #{course_number}"
+    if subdepartment
```
Comment (aw3as) [inline app/models/course.rb]: are there any courses without a subdepartment??

File: app/views/courses/show.html.slim [hunk 8]
```diff
@@ -1,6 +1,9 @@
 - content_for :page_heading, "#{@course.mnemonic_number} - #{@course.title}"
+- content_for :course_link, "/courses/#{@course.id}/professors"
```
Comment (aw3as) [inline app/views/courses/show.html.slim]: are these styled?

File: app/views/search/search.html.slim [hunk 9]
```diff
@@ -1,32 +1,41 @@
...
-      .row
+  - if @count < 1
+    .col-xs-12
+      = "No results for #{@query} :("
+  - else
+    - if @departments.count > 0
+      h2 = pluralize(@departments.count, "Department")
+      - @departments.each do |department|
+        .row
+          a href="/departments/#{department[:id]}" data-no-turbolink=""
```
Comment (aw3as) [inline app/views/search/search.html.slim]: why disable turbolink?

File: app/models/course.rb (hunk 7 above)
Comment (emumoo) [inline app/models/course.rb]: 2..

File: app/views/search/search.html.slim (hunk 9 above)
Comment (emumoo) [inline app/views/search/search.html.slim]: idk, copy/pasted from old search page links

File: app/assets/stylesheets/professor.scss (hunk 4 above)
Comment (emumoo) [inline app/assets/stylesheets/professor.scss]: when search returns professor, it links to this

File: app/assets/stylesheets/application.scss (hunk 1 above)
Comment (emumoo) [inline app/assets/stylesheets/application.scss]: "go back" links in course page
course title takes you to courses/[id]/professors
professor title takes you to professor page

I think @keenadams wants to implement breadcrumbs at some point but for now,
this should make navigation easier

File: app/views/courses/show.html.slim (hunk 8 above)
Comment (emumoo) [inline app/views/courses/show.html.slim]: yes

**Pull request merged**